src/V3Delayed.cpp

```diff
@@ -279,6 +431,25 @@ class DelayedVisitor final : public VNVisitor {
 
         // In a suspendable of fork, we must use the unique flag scheme, TODO: why?
         if (vscpInfo.m_inSuspOrFork) return Scheme::FlagUnique;
+
+        const bool isIntegralOrPacked = dtypep->isIntegralOrPacked();
+        // Check for mixed usage (this also warns if not OK)
+        if (checkMixedUsage(vscp, isIntegralOrPacked)) {
+            // If it's a variable updated by both blocking and non-blocking
+            // asignments, use the ShadowVarMasked schem if masked update is
+            // possible. This can handle blocking and non-blocking updates to
+            // inpdendent parts correctly at run-time, and always works, even
+            // in loops or other dynamic context.
+            if (isIntegralOrPacked) return Scheme::ShadowVarMasked;
+            // If it's inside a loop, use Scheme::ShadowVar, which is safe,
+            // but will generate incorrect code if a partial update is used
+            if (vscpInfo.m_inLoop) return Scheme::ShadowVar;
```

Comment:
Strictly speaking the case of "mixed assignments, with the NBA in a loop (or other dynamic context), and the target being a non-packedOrIntegral type, still doesn't work safely. I added this case here which safely handles whole (non-partial) updates in loops, and yields invalid C++ when a partial udpate is used. I think this is the same as we used to do before. Given there are other exotic cases that silently don't work properly, this is probably ok without an error for now?

Reply:
All these cases need https://github.com/verilator/verilator/pull/5092#issuecomment-2090125582